backend/src/dao/DevPortfolioDao.ts

```diff
@@ -75,26 +60,45 @@ export default class DevPortfolioDao {
 
     const data = doc.data() as DBDevPortfolio;
 
-    return DevPortfolioDao.DBDevPortfolioToDevPortfolio(data, true, null);
+    return DevPortfolioDao.DBDevPortfolioToDevPortfolio(data);
   }
 
-  static async getAllInstances(
-    isAdminReq: boolean,
-    user: IdolMember | null
-  ): Promise<DevPortfolio[]> {
+  static async getAllInstances(): Promise<DevPortfolio[]> {
     const instanceRefs = await devPortfolioCollection.get();
 
     return Promise.all(
       instanceRefs.docs.map(async (instanceRefs) =>
-        DevPortfolioDao.DBDevPortfolioToDevPortfolio(
-          instanceRefs.data() as DBDevPortfolio,
-          isAdminReq,
-          user
-        )
+        DevPortfolioDao.DBDevPortfolioToDevPortfolio(instanceRefs.data() as DBDevPortfolio)
       )
     );
   }
 
+  public static async getAllDevPortfolioInfo(): Promise<DevPortfolioInfo[]> {
+    const instanceInfoRefs = await devPortfolioCollection
+      .select('deadline', 'earliestValidDate', 'name', 'uuid', 'lateDeadline')
+      .get();
+    return Promise.all(
+      instanceInfoRefs.docs.map(async (instanceRefs) => instanceRefs.data() as DevPortfolioInfo)
+    );
+  }
+
+  public static async getDevPortfolioInfo(uuid: string): Promise<DevPortfolioInfo> {
+    const portfolioRef = await devPortfolioCollection.doc(uuid).get();
+    return portfolioRef.data() as DevPortfolioInfo;
+  }
+
+  public static async getUsersDevPortfolioSubmissions(
+    uuid: string,
+    user: IdolMember
+  ): Promise<DevPortfolioSubmission[]> {
+    const portfolioData = (await devPortfolioCollection.doc(uuid).get()).data() as DBDevPortfolio;
+    const dBSubmissions = portfolioData.submissions.filter(
+      (submission) => submission.member.id === user.email
+    );
```

Comment:
Just want to double check to make sure that we're able to do equality comparisons with `===` for firestore document references.

Reply:
This does work.